command/agent/config.go

```diff
@@ -1612,7 +1612,8 @@ func mergeVaultConfigs(left, right map[string]*config.VaultConfig) map[string]*c
 		if lConfig, ok := left[name]; ok {
 			merged[name] = lConfig.Merge(rConfig)
 		} else {
-			merged[name] = rConfig.Copy()
+			merged[name] = config.DefaultVaultConfig().Merge(rConfig).Copy()
```

Comment:
Or maybe `config.DefaultVaultConfig().Merge(rConfig.Copy())`? Does it actually make a difference? 😅

Reply:
I don't think we need the `Copy()` at all if we're calling `Merge`, because the field values are being copied from `rConfig` onto of the default config.